client/client.go

```diff
@@ -423,16 +482,8 @@ const (
 	ErrorKindNoDefaultServices = "no-default-services"
 )
 
-func (rsp *response) err(cli *Client) error {
-	if cli != nil {
-		maintErr := rsp.Maintenance
-		// avoid setting to (*client.Error)(nil)
-		if maintErr != nil {
-			cli.maintenance = maintErr
-		} else {
-			cli.maintenance = nil
-		}
-	}
```

Comment:
This is moved to the ```decoder(...)``` implementation, but still happens exactly at the same point, just before the error is unmarshalled.